src/components/Button/Button.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: React.ReactNode;
+  name: string;
+}
```

Comment:
I would suggest here going with `interface ButtonProps extends React.ComponentProps<'button'> {}` instead of extending `React.ButtonHTMLAttributes<HTMLButtonElement>`.

https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element

The current solution throws an error when we pass in a `ref` and since this is a highly reusable component, we need to support forwarding refs to it. I personally also find `ComponentProps` more readable. There is also no need to redefine the `children` and `name` props since they are already included in the type we are extending.

I personally also prefer using type aliases instead of interfaces when working on a user-facing application like this one to prevent the cases where another developer would redefine the interface `ButtonProps` and it would automatically merge these types to it without throwing an error. I would use a type alias like this instead, but it is also fine if you would prefer to keep it as an interface.

```ts
type ButtonProps = ComponentProps<'button'>;
```